packages/legend-extension-dsl-serializer/src/components/DSLSerializer_StudioPlugin.tsx

```diff
@@ -0,0 +1,146 @@
+  ElementProjectExplorerDnDTypeGetter,
+  ElementIconGetter,
+  DSL_StudioPlugin_Extension,
+  NewElementState,
+} from '@finos/legend-studio';
+import { FileIcon } from '@finos/legend-art';
+import { SchemaSetEditorState } from '../stores/SchemaSetEditorState';
+import { BindingEditorState } from '../stores/BindingEditorState';
+import type { PackageableElement } from '@finos/legend-graph';
+import { SchemaSet } from '../models/metamodels/pure/model/packageableElements/schemaSet/SchemaSet';
+import { Binding } from '../models/metamodels/pure/model/packageableElements/store/Binding';
+
+const SCHEMA_SET_ELEMENT_TYPE = 'SCHEMASET';
+const SCHEMA_SET_ELEMENT_PROJECT_EXPLORER_DND_TYPE =
+  'PROJECT_EXPLORER_SCHEMA_SET';
+const BINDING_ELEMENT_TYPE = 'BINDING';
+const BINDING_ELEMENT_PROJECT_EXPLORER_DND_TYPE = 'PROJECT_EXPLORER_BINDING';
+
+export class DSLSerializer_StudioPlugin
+  extends StudioPlugin
+  implements DSL_StudioPlugin_Extension
+{
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+  }
+
+  install(pluginManager: StudioPluginManager): void {
+    pluginManager.registerStudioPlugin(this);
+  }
+
+  getExtraSupportedElementTypes(): string[] {
+    return [SCHEMA_SET_ELEMENT_TYPE, BINDING_ELEMENT_TYPE];
+  }
+
+  getExtraElementTypeGetters(): ElementTypeGetter[] {
+    return [
+      (element: PackageableElement): string | undefined => {
+        if (element instanceof SchemaSet) {
+          return SCHEMA_SET_ELEMENT_TYPE;
+        } else if (element instanceof Binding) {
+          return BINDING_ELEMENT_TYPE;
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getExtraElementIconGetters(): ElementIconGetter[] {
+    return [
+      (type: string): React.ReactNode | undefined => {
+        if (type === SCHEMA_SET_ELEMENT_TYPE) {
+          return (
+            <div className="icon icon--text-element">
+              <FileIcon />
+            </div>
+          );
+        } else if (type === BINDING_ELEMENT_TYPE) {
+          return (
+            <div className="icon icon--text-element">
+              <FileIcon />
+            </div>
+          );
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getExtraNewElementFromStateCreators(): NewElementFromStateCreator[] {
+    return [
+      (
+        type: string,
+        name: string,
+        state: NewElementState,
+      ): PackageableElement | undefined => {
+        if (type === SCHEMA_SET_ELEMENT_TYPE) {
+          return new SchemaSet(name);
+        } else if (type === BINDING_ELEMENT_TYPE) {
+          return new Binding(name);
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getExtraElementEditorStateCreators(): ElementEditorStateCreator[] {
+    return [
+      (
+        editorStore: EditorStore,
+        element: PackageableElement,
+      ): ElementEditorState | undefined => {
+        if (element instanceof SchemaSet) {
+          return new SchemaSetEditorState(editorStore, element);
```

Comment:
Technically, if we don't support form editor for these elements, you can just use `UnsupportedElementEditorState`